src/native/libs/System.Native/pal_process.c

```diff
@@ -500,10 +500,22 @@ done:;
             waitpid(processId, &status, 0);
         }
 
-        *stdinFd = -1;
-        *stdoutFd = -1;
-        *stderrFd = -1;
-        *childPid = -1;
+        if (stdinFd != NULL)
```

Comment:
It would be better to change the checks for null arguments at the top of the function to assert instead.

The libraries under src/native/libs are only used as an internal implementation detail of the managed libraries in this repo. We can trust the libraries in this repo that they are going not going to pass in bogus arguments.

Reply:
Are you going to change the error handling at the top of the method to assert instead of return EINVAL? I think it would be a good idea so that the static analyzers stop complaining.

Reply:
My comment somehow disappeared. There is null check at the beginning so it is a false alarm from the system. So return to the original code.
https://github.com/dotnet/runtime/blob/b0920912512aa7e205363b046a1911e6a97df31f/src/native/libs/System.Native/pal_process.c#L247-L255

Reply:
Right, I am suggesting to change the check at the beginning to `assert (NULL != filename && NULL != argv && NULL != envp ...);` so that there is no false alarm.

Reply:
Oh I see. Applied the suggestion. Thank you.